### PR TITLE
fix: building issue under the Darwin and Windows

### PR DESCRIPTION
**Pull request by mervinkid**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

fix building issue under the Darwin and Windows operating system.

notes:
> Building on un-linux system is just for debugging and study. Linux kernel based operating system is required for full features.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by mervinkid** [comment]
All checks have passed.

File: storage/fallocate_darwin.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"syscall"
+
+	"github.com/chubaofs/chubaofs/util/log"
+)
+
+func fallocate(fd int, mode uint32, off int64, len int64) (err error) {
+	// system call 'fallocate' is not supported in Darwin(Apple MacOS).
+	log.LogWarnf("fallocate: not supported in Darwin(Apple MacOS) operating system")
```
Comment (iliul) [inline storage/fallocate_darwin.go]: [go-fallocate](https://github.com/detailyang/go-fallocate/blob/master/fallocate_darwin.go#L9)  implement syscall "fallocate" with different os. How about import it here？

Comment (mervinkid) [reply storage/fallocate_darwin.go]: Thank you for your suggestion.
We will follow up and test your proposal.

Comment (mervinkid) [reply storage/fallocate_darwin.go]: > [go-fallocate](https://github.com/detailyang/go-fallocate/blob/master/fallocate_darwin.go#L9) implement syscall "fallocate" with different os. How about import it here？

Thanks in advance for your suggestions.
But your proposal does not match our requirement.

We use `fallocate` system call with **hole punching** flag to release disk space used by unused tiny files. 

The `go-fallocate` can not make possible in hole punching.

More detail about our storage engine  see our paper [CFS: A Distributed File System for Large Scale Container Platforms](https://arxiv.org/abs/1911.03001).

Comment (iliul) [reply storage/fallocate_darwin.go]: Yep, I see. Will do it later.

**Pull request merged**